app/models/app_feature_completion_migration.rb

```diff
@@ -0,0 +1,57 @@
+class AppFeatureCompletionMigration
+  include ActiveModel::Validations
+
+  attr_reader :app, :feature_gate, :version_before_type_cast
+
+  validate :feature_gate_must_exist
+
+  delegate :version, to: :app_feature_completion
+
+  def initialize(params)
```

Comment:
Any reason not to do kwargs here?

Reply:
yeah, they're hostile to StrongParameters, I believe? This thing is an immutable resource model effectively.